server/src/main/java/io/druid/server/initialization/jetty/JettyServerModule.java

```diff
@@ -253,6 +255,17 @@ public void start() throws Exception
           @Override
           public void stop()
           {
+            if (activeConnections.get() != 0) {
+              long graceFullShutDownTime = (int)server.getAttribute(GRACEFUL_SHUTDOWN_TIME);
+              log.info("Waiting for [%s] milliseconds for active requests to be zero, current active requests=[%s]",
+                  graceFullShutDownTime, activeConnections.get());
+              try {
+                Thread.sleep(graceFullShutDownTime);
+              } catch (InterruptedException e) {
+                log.error("Sleep has been interrupted, while waiting for active requests to be zero");
```

Comment:
`Thread.currentThread().interrupt();` maybe and/or rethrow

Reply:
Thanks. Addressed it.